java-checks/src/main/java/org/sonar/java/checks/ServletMethodsExceptionsThrownCheck.java

```diff
@@ -95,6 +97,17 @@ private static List<Type> getCaughtExceptions(List<CatchTree> catches) {
   }
 
   private void checkMethodInvocation(MethodInvocationTree node) {
+    if (node.methodSelect() instanceof MemberSelectExpressionTree memberSelect
+      && memberSelect.expression() instanceof IdentifierTree identifier && "Try".equals(identifier.name())
+      && "run".equals(memberSelect.identifier().name())) {
```

Comment:
It looks like the same behavior could also happen for `runRunnable. ' It's better to cover it as well. At that point, I think it's better to simplify this if statement, that has too many conditions, by extracting it in a nice method